engine/dlinked_list.hpp

```diff
@@ -191,26 +192,26 @@ class DLinkedList {
               DLRecord *tail_pmmptr)
       : pmem_allocator_ptr_{pmem_allocator_p}, head_pmmptr_{head_pmmptr},
         tail_pmmptr_{tail_pmmptr} {
+
 #if DEBUG_LEVEL >= 0
     {
       kvdk_assert(head_pmmptr->entry.meta.type == HeadType,
                   "Cannot rebuild a DlinkedList from given PMem pointer "
                   "not pointing to a valid Head Record!");
       iterator curr{pmem_allocator_ptr_, head_pmmptr};
       ++curr;
-
       while (true) {
         switch (static_cast<RecordType>(curr->entry.meta.type)) {
         case DataType: {
           iterator next{curr};
           ++next;
-          iterator prev{curr};
-          --prev;
           PMemOffsetType offset_curr = curr.GetCurrentOffset();
-          kvdk_assert(next->prev == offset_curr,
-                      "Found broken linkage when rebuilding DLinkedList!");
-          kvdk_assert(prev->next == offset_curr,
-                      "Found broken linkage when rebuilding DLinkedList!");
+          if (next->prev != offset_curr) {
+            GlobalLogger.Error(
+                "Found broken linkage when rebuilding DLinkedList! Repaired");
+            next->prev = offset_curr;
+            pmem_persist(&next->prev, sizeof(PMemOffsetType));
```

Comment:
need to repair? why?

Reply:
If the program is terminated abnormally, there's a small chance that a update, insertion or deletion is not completed. This happens in Skiplist, too.